dependencies.gradle

```diff
@@ -0,0 +1,18 @@
+// Add your dependencies here
+
+dependencies {
+    compile("com.github.GTNewHorizons:AppleCore:master-SNAPSHOT:dev") {
+        transitive=false;
```

Comment:
`compile` dependencies are used for compilation and runtime. So transitive dependencies for execution are required. `transitive = false` must be set for `compileOnly` and for `compileOnly` alone. It is set by default, but forge Gradle did some pre-gradle 2 stuff and overrides it. Might change that in the future.

Reply:
Meh, it complains about missing spongemixins without this, since applecore needs it.
```
Execution failed for task ':compileJava'.
> Could not resolve all files for configuration ':compileClasspath'.
   > Could not find org.spongepowered:mixin:0.7.11-SNAPSHOT.
```

Reply:
when I add in the spongemixin repo, I get this on runClient

```
cpw.mods.fml.common.LoaderException: java.lang.ExceptionInInitializerError
        at cpw.mods.fml.common.LoadController.transition(LoadController.java:163)
        at cpw.mods.fml.common.Loader.preinitializeMods(Loader.java:559)
        at cpw.mods.fml.client.FMLClientHandler.beginMinecraftLoading(FMLClientHandler.java:243)
        at net.minecraft.client.Minecraft.startGame(Minecraft.java:480)
        at net.minecraft.client.Minecraft.run(Minecraft.java:878)
        at net.minecraft.client.main.Main.main(SourceFile:148)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
        at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
        at net.minecraftforge.gradle.GradleStartCommon.launch(Unknown Source)
        at GradleStart.main(Unknown Source)
Caused by: java.lang.ExceptionInInitializerError
        at squeek.applecore.AppleCore.preInit(AppleCore.java:39)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at cpw.mods.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:532)
        at sun.reflect.GeneratedMethodAccessor5.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
        at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
        at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
        at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
        at com.google.common.eventbus.EventBus.post(EventBus.java:275)
        at cpw.mods.fml.common.LoadController.sendEventToModContainer(LoadController.java:212)
        at cpw.mods.fml.common.LoadController.propogateStateMessage(LoadController.java:190)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
        at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
        at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
        at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
        at com.google.common.eventbus.EventBus.post(EventBus.java:275)
        at cpw.mods.fml.common.LoadController.distributeStateMessage(LoadController.java:119)
        at cpw.mods.fml.common.Loader.preinitializeMods(Loader.java:556)
        ... 12 more
Caused by: cpw.mods.fml.relauncher.ReflectionHelper$UnableToFindFieldException: java.lang.NoSuchFieldException: starveTimer
        at cpw.mods.fml.relauncher.ReflectionHelper.findField(ReflectionHelper.java:94)
        at squeek.applecore.api_impl.AppleCoreAccessorMutatorImpl.<clinit>(AppleCoreAccessorMutatorImpl.java:240)
        ... 39 more
Caused by: java.lang.NoSuchFieldException: starveTimer
        at java.lang.Class.getDeclaredField(Class.java:2070)
        at cpw.mods.fml.relauncher.ReflectionHelper.findField(ReflectionHelper.java:85)
        ... 40 more


```

Reply:
A known issue. Apple core will get some love when I'm back. The setup is correct that way though. Can you leave it `compileOnly`? Apple core never ran in dev anyway...

Reply:
that explains why I had it in the run/mods folder.  Though it ran in dev now; will dig into later.